substrate/client/network/sync/src/strategy/warp.rs

```diff
@@ -487,7 +499,17 @@ where
 		let peer_id = self.schedule_next_peer(PeerState::DownloadingProofs, None)?;
 		trace!(target: LOG_TARGET, "New WarpProofRequest to {peer_id}, begin hash: {begin}.");
 
-		Some((peer_id, WarpProofRequest { begin }))
+		let request = WarpProofRequest { begin };
+
+		let Some(protocol_name) = self.protocol_name.clone() else {
+			warn!(
```

Comment:
nit: Wouldn't this generate too much noise?

Reply:
It will not, this method is not supposed to be called when there is no warp sync in the first place, that would basically be an implementation bug. This code effectively migrated from `substrate/client/network/sync/src/engine.rs` (see the change at the bottom of the file). Should be clear if looking at individual commits.